tests/rustdoc/sidebar/sidebar-links-to-foreign-impl.rs

```diff
@@ -7,10 +7,12 @@
 //@ has - '//h2[@id="foreign-impls"]' 'Implementations on Foreign Types'
 //@ has - '//*[@class="sidebar-elems"]//section//a[@href="#impl-Foo-for-u32"]' 'u32'
 //@ has - '//*[@id="impl-Foo-for-u32"]//h3[@class="code-header"]' 'impl Foo for u32'
+//@ has - '//*[@id="impl-Foo-for-u8"]//h3[@class="code-header"]' 'impl Foo for u8'
```

Comment:
how to assert `u8` and `u32` is in a sorted way ...

Reply:
We can check if it's the nth item of this type, but not sure we can outdide gui tests...

Reply:
I just added new ui testcase:

`tests/rustdoc/sidebar-link-sort.rs` and `tests/rustdoc/sidebar-link-sort.sidebar.html`